### PR TITLE
Fixed #11560 -- Allowed proxy model multiple-inheritance from the same concrete base model.

### PR DESCRIPTION
**Pull request by akki**
Fixed [#11560](https://code.djangoproject.com/ticket/11560/).

### CODE REVIEW

File: django/db/models/base.py [hunk 1]
```diff
@@ -177,10 +177,10 @@ def __new__(cls, name, bases, attrs):
                         )
                     else:
                         continue
-                if base is not None:
-                    raise TypeError("Proxy model '%s' has more than one non-abstract model base class." % name)
-                else:
+                if base is None:
                     base = parent
+                elif parent._meta.concrete_model is not base._meta.concrete_model:
+                    raise TypeError("Proxy model '%s' has more than one non-abstract model base class." % name)
```
Comment (timgraham) [inline django/db/models/base.py]: No tests fail if I remove this exception - could you try to add one?

File: docs/topics/db/models.txt [hunk 2]
```diff
@@ -1232,6 +1232,11 @@ provide any connection between the rows in the different database tables. A
 proxy model can inherit from any number of abstract model classes, providing
 they do *not* define any model fields.
 
+.. versionchanged:: 1.10
```
Comment (timgraham) [inline docs/topics/db/models.txt]: Put this annotation below the paragraph and describe the behavior in older versions below it.

**Comment by akki** [comment]
Updated the patch according to the above suggestions.

File: tests/proxy_models/tests.py [hunk 3]
```diff
@@ -105,6 +105,17 @@ def test_proxy_included_in_ancestors(self):
         with self.assertRaises(Person.MultipleObjectsReturned):
             StatusPerson.objects.get(id__lt=max_id + 1)
 
+    def test_multiple_abc(self):
```
Comment (timgraham) [inline tests/proxy_models/tests.py]: @charettes, do we need

File: tests/proxy_models/tests.py [hunk 4]
```diff
@@ -105,6 +105,17 @@ def test_proxy_included_in_ancestors(self):
...
+        """
+        Proxy cannot have multiple abstract base classes
+        """
+        def build_multiple_abc():
```
Comment (timgraham) [inline tests/proxy_models/tests.py]: I could be missing something, but I don't see a need for the function.
```
msg = "Proxy model 'MultipleAbstract' has more than one non-abstract model base class."
with self.assertRaisesMessage(TypeError, msg):
    class MultipleAbstract(User, Person):
        class Meta:
            proxy = True
```
should be okay, I think? If so, we can make a separate commit to clean up the existing tests (and make them use `assertRaisesMessage` instead of just `assertRaises` too.

File: tests/proxy_models/tests.py (hunk 3 above)
Comment (charettes) [inline tests/proxy_models/tests.py]: No as `User` and `Person` are registered to the global `apps` instance. The creation of the class is supposed to fail anyway so the `apps` cache won't get polluted.

Comment (timgraham) [inline tests/proxy_models/tests.py]: Can it be removed from the existing tests `test_no_base_classes` and `test_no_cbc` in this file (https://github.com/django/django/pull/6360)?

File: tests/proxy_models/tests.py (hunk 4 above)
Comment (timgraham) [inline tests/proxy_models/tests.py]: (done in https://github.com/django/django/pull/6360)

**Comment by timgraham** [comment]
merged in 49f95cc0a0167e91fd0f6987428905911bff076c, thanks!
